packages/server/src/watch.ts

```diff
@@ -0,0 +1,32 @@
+import {checkMatch} from "@prodo-ai/snoopy-search";
+import * as chokidar from "chokidar";
+import {generateComponentsFileContents} from "./generate";
+
+export const watchComponentsFile = async (
+  cwd: string,
+  onComponentsFileChange: () => void,
+): Promise<chokidar.FSWatcher> => {
+  let generated = await generateComponentsFileContents(".", cwd);
+
+  return chokidar
+    .watch(".", {
+      cwd,
+      ignoreInitial: true,
+      ignored: /node_modules|\.git|flycheck_*/,
```

Comment:
it is half duplicated, although they use different syntax. In search the list of stuff to ignore is

```ts
export const exampleFileGlob = [
  `**/*.{${fileExtensions.join(",")}}`,
  "!**/flycheck_*.*",
  "!node_modules/**/*",
  "!**/test/**/*",
  "!**/*.test.*",
];
```

it doesn't include git because `globby` ignore dotfiles automatically. Also, the format is not regex and a file glob instead. Ill try and combine and hopefully it is not too messy